### ISSUE TITLE
About input camera convention.

### ISSUE THREAD

**Issue by Badi-Li** [ByteDance-Seed/Depth-Anything-3#158]
Hi, thanks for the great work!
I want to check if the input camera follow opencv convention and should be world-to-camera? As I found the reconstruction with camera condition looks worse than the pure images ones. So I just want to check if my camera convention is correct. I actually use the following function to get extrinsics and intrinsics from yaw, pitch, radius and fov:
```python
def yaw_pitch_r_fov_to_extrinsics_intrinsics(yaws, pitchs, rs, fovs):
    is_list = isinstance(yaws, list)
    if not is_list:
        yaws = [yaws]
        pitchs = [pitchs]
    if not isinstance(rs, list):
        rs = [rs] * len(yaws)
    if not isinstance(fovs, list):
        fovs = [fovs] * len(yaws)
    extrinsics = []
    intrinsics = []
    for yaw, pitch, r, fov in zip(yaws, pitchs, rs, fovs):
        fov = torch.deg2rad(torch.tensor(float(fov))).cuda()
        yaw = torch.deg2rad(torch.tensor(float(yaw))).cuda()
        pitch = torch.deg2rad(torch.tensor(float(pitch))).cuda()
        orig = torch.tensor([
            torch.sin(yaw) * torch.cos(pitch),
            torch.cos(yaw) * torch.cos(pitch),
            torch.sin(pitch),
        ]).cuda() * r
        extr = utils3d.torch.extrinsics_look_at(orig, torch.tensor([0, 0, 0]).float().cuda(), torch.tensor([0, 0, 1]).float().cuda())
        intr = utils3d.torch.intrinsics_from_fov_xy(fov, fov)
        extrinsics.append(extr)
        intrinsics.append(intr)
    if not is_list:
        extrinsics = extrinsics[0]
        intrinsics = intrinsics[0]
    return extrinsics, intrinsics
```
where the utils3d.torch.extrinsics_look_at is defined as
```python
@batched(1, 1, 1)
def extrinsics_look_at(
    eye: torch.Tensor,
    look_at: torch.Tensor,
    up: torch.Tensor
) -> torch.Tensor:
    """
    Get OpenCV extrinsics matrix looking at something

    Args:
        eye (torch.Tensor): [..., 3] the eye position
        look_at (torch.Tensor): [..., 3] the position to look at
        up (torch.Tensor): [..., 3] head up direction (-y axis in screen space). Not necessarily othogonal to view direction

    Returns:
        (torch.Tensor): [..., 4, 4], extrinsics matrix
    """
    N = eye.shape[0]
    z = look_at - eye
    x = torch.cross(-up, z, dim=-1)
    y = torch.cross(z, x, dim=-1)
    # x = torch.cross(y, z, dim=-1)
    x = x / x.norm(dim=-1, keepdim=True)
    y = y / y.norm(dim=-1, keepdim=True)
    z = z / z.norm(dim=-1, keepdim=True)
    R = torch.stack([x, y, z], dim=-2)
    t = -torch.matmul(R, eye[..., None])
    ret = torch.zeros((N, 4, 4), dtype=eye.dtype, device=eye.device)
    ret[:, :3, :3] = R
    ret[:, :3, 3] = t[:, :, 0]
    ret[:, 3, 3] = 1.
    return ret
```
I further scale the intrinsics accroding to the rendering resolution as 
```python
intrinsics[..., 0, :] *= W 
intrinsics[..., 1, :] *= H
```
With this camera input, unfortunately I got low reconstruction quality:
<img width="227" height="335" alt="Image" src="https://github.com/user-attachments/assets/ee31dea4-b424-46cf-aa7d-5239aed17369" />

But without camera condition, it somehow better:
<img width="231" height="290" alt="Image" src="https://github.com/user-attachments/assets/fa190136-7850-4e95-9a71-6bd2ad3376b5" />

I also tried to apply `affine_inverse(extrinsics)`, but this time I can visualize nothing.

**Comment by Golbstein** [comment]
the convention is w2c
you can see this in their extrinsic normalization step at `_normalize_extrinsics`

you need to provide unnormalized w2c and make sure you have a GT positions and not "approximate" positions.
also, and important - make sure your camera convention at COLMAP\Opencv i.e. X=Right, Y=Down, Z=Forward

sometimes extrinsic in various dataset represented in a different coordinate system. so for instace, when your coord system is: X=Forward, Y=Left, Z=Up you have to transform your extrinsic in that way:

```python
T_current2colmap = np.array([[0, 0, 1, 0], [-1, 0, 0, 0], [0, -1, 0, 0], [0, 0, 0, 1]])
c2w_da3 = c2w @ T_current2colmap
extrinsic = np.linalg.inv(c2w_da3)
```

**Comment by libd1** [comment]
Many thanks to ur suggestions and help! 
Actually, I have checked the two functions I provided, to my understanding they return a w2c matrix following X=Right, Y=Down, Z=Forward. And I'm sure the positions are GT because I actually used the extrinsics sampled, to render a 3DGS, to get the images. I would keep checking other potential issues, and again, appreciate ur help!

**Comment by Badi-Li** [comment]
I finally found that the issue is about the files formatting, and the camera pose obtained by the lookat function is actually the correct opencv w2c camera.

**Comment by kevinhuangxf** [comment]
> I finally found that the issue is about the files formatting, and the camera pose obtained by the lookat function is actually the correct opencv w2c camera.

Hi. May I ask whether your issue of the input camera poses solved? I also encounter of using camera pose to generate bad looking results. While without using the camera poses the results look much better.
